File: tests/java/com/trulioo/sdk/model/ConsentTest.java
package com.trulioo.sdk.model;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.*;

public class ConsentTest {
    private Consent consent;

    @BeforeEach
    public void beforeEach() {
        consent = new Consent();
    }

    @Test
    public void testName() {
        String name = "test";
        consent.setName(name);

        assertEquals(name, consent.getName());
        assertEquals(name, new Consent().name(name).getName());
    }

    @Test
    public void testText() {
        String text = "desc";
        consent.setText(text);

        assertEquals(text, consent.getText());
        assertEquals(text, new Consent().text(text).getText());
    }

    @Test
    public void testUrl() {
        String url = "https://trulioo.com";
        consent.setUrl(url);

        assertEquals(url, consent.getUrl());
        assertEquals(url, new Consent().url(url).getUrl());
    }

    @Test
    public void testEquals() {
        String name = "test";
        Consent consent1 = new Consent().name(name);

        assertEquals(consent1, consent1);
        assertEquals(consent1, new Consent().name(name));
        assertNotEquals(consent1, new Consent().name(name + "1"));
        assertNotEquals(consent1, null);
    }

    @Test
    public void testHashcode() {
        String name = "test";

        assertEquals(new Consent().name(name).hashCode(), new Consent().name(name).hashCode());
    }

    @Test
    public void testToString() {
        String name = "test";

        assertEquals(new Consent().name(name).toString(), new Consent().name(name).toString());
    }
}

File: tests/php/Model/ConsentTest.php
<?php

namespace Trulioo\SDK\Test\Model;

use PHPUnit\Framework\TestCase;
use Trulioo\SDK\Model\Consent;

class ConsentTest extends TestCase
{
    private Consent $consent;

    public function setUp(): void
    {
        $this->consent = new Consent([]);
    }

    public function testPropertyName()
    {
        $this->consent->setName('Credit Agency');
        $this->assertEquals('Credit Agency', $this->consent->getName());
    }

    public function testPropertyText()
    {
        $this->consent->setText('A credit agency');
        $this->assertEquals('A credit agency', $this->consent->getText());
    }

    public function testPropertyUrl()
    {
        $this->consent->setUrl('https://trulioo.com');
        $this->assertEquals('https://trulioo.com', $this->consent->getUrl());
    }
}
